rbi/stdlib/csv.rbi

```diff
@@ -267,13 +267,20 @@ class CSV < Object
   # would read UTF-32BE data from the file but transcode it to UTF-8 before
   # [`CSV`](https://docs.ruby-lang.org/en/2.6.0/CSV.html) parses it.
   sig do
-    type_parameters(:U).params(
+    params(
         path: T.any(String, ::Sorbet::Private::Static::IOLike),
-        options: T::Hash[Symbol, T.type_parameter(:U)],
-        blk: T.proc.params(arg0: T::Array[T.nilable(String)]).void,
+        options: T::Hash[Symbol, BasicObject],
+        blk: T.proc.params(arg0: T.any(T::Array[T.nilable(BasicObject)], CSV::Row)).void,
```

Comment:
Would you mind changing this here and in `parse` to `T.any(T::Array[T.untyped], CSV::Row))`?

Consider the following example:

```ruby
total = 0
CSV.parse(file_content, {converters: %i[numeric]}) do |row|
  col1, _ = row
  total += col1
end
```

Sorbet doesn't have any way of knowing what the return types of the `converters` are. With your change, there would be a static type error on `total += col1`, because you can't add `Integer` and `BasicObject`. Also, given that the type is specified as `BasicObject`, you can't just call `to_i` on it, so you are left with having to `T.cast` it.

Reply:
ah, good point, haven't thought that through. I will have some time tomorrow to address it. Thanks a lot for looking into it

Reply:
actually, @marianosimone quick question for you: would you prefer to change other sigs as well, given that `by_col` and other methods would return the values that could have been processed with `converters`?

Reply:
Yes, unfortunately we'll have to go with `T.untyped` there as well, thanks for pointing that out!